decompiler_agent/src/main/java/org/jrd/agent/InstrumentationProvider.java

```diff
@@ -27,10 +27,9 @@
         
     public void setClassBody(String cname, byte[] nwBody) throws UnmodifiableClassException {
         Class clazz = findClass(cname);
-        transformer.allowToSaveBytecode(); //?
+        transformer.allowToSaveBytecode();
         try {
             String nameWithSlashes = clazz.getName().replace(".", "/");
-            transformer.allowToSaveBytecode(); //? taken from getClassBody. Really twice?
```

Comment:
How had you verified this? Isnt the placement in try block better option?

Reply:
Yes, I was looking down the function and it's only setting variable to true. And I tried it too.

Reply:
If it is in the try block it is't working

Reply:
Interesting. Thank you!